ui-v2/app/utils/get-object-pool.js

```diff
@@ -0,0 +1,52 @@
+export default function(dispose = function() {}, max, objects = []) {
```

Comment:
Seems like an odd design to allow the object array to be provided. It opens up the opportunity for it be mucked with externally.

Reply:
It means I can inspect it whilst testing:

https://github.com/hashicorp/consul/pull/5083/files/4a22886de3d6e75262b4bc71ae5b7b1bec61fb49#diff-876e4c1f4e8e565beb60f501df6c19d1R18

You could pass your own array and then muck with it externally, there are lots of things you can do but shouldn't. I don't think we pass an array in here anywhere in the application code (only during testing). Maybe I should add some comments here to make it clearer?